### ISSUE TITLE
When quick open is configured to include symbols, slow symbol results block file results

### ISSUE THREAD

**Issue by nicksnyder** [microsoft/vscode#27152]
**Problem**

If `search.quickOpen.includeSymbols` is configured to true then OpenAnythingHandler will wait for `this.openSymbolHandler.getResults(searchValue)` to resolve. This is a problem if symbol results are slow because the OpenAnythingHandler is the only way to get file results.

**Ideal solution**

Stream file and symbol results in as they are found so the user is not blocked on slow results.

**Alternative solution**

Don't wait indefinitely for symbol results in the OpenAnythingHandler. If the results timeout, the user can always perform the dedicated OpenSymbolHandler quick open action.

**Comment by nicksnyder** [comment]
I created a PR that implements the alternative solution for the sake of discussion: #27153

**Comment by chrmarti** [comment]
Having a setting to control the timeout feels like making the problem worse, because it does solve anything (what would be a sensible value?) and instead adds a setting that seems hard to use. I suggest we close the PR.

**Comment by nicksnyder** [comment]
How about this? https://github.com/Microsoft/vscode/pull/27207

**Comment by nicksnyder** [comment]
Thanks for merging #27207 !

**Comment by chrmarti** [comment]
@nicksnyder I've added a change that fixes edge cases in the controller, let me know if that works for you. Thanks for your contribution!

**Comment by nicksnyder** [comment]
Will pull and test out, thanks!

**Comment by nicksnyder** [comment]
Works for me!

**Comment by nicksnyder** [comment]
@chrmarti Hmm, after planing around with this more today I am seeing some weird non-deterministic behavior around results not showing up when they should be. I haven't had a chance to dig in to it yet but wanted to give you a heads up. Might want to revert out the change to be safe.

**Comment by chrmarti** [comment]
@nicksnyder Do you have steps to reproduce?

**Comment by nicksnyder** [comment]
No, sadly another eng reported the issue after I left for the day. It also possible that this change is fine and the issue I am seeing is caused by other modifications in our fork. Just wanted to give you a heads up in case you see any weird behavior. I will investigate more next week.

**Comment by sqs** [comment]
I work with @nicksnyder and looked into this some more. I don't think it's a critical issue for VS Code, because it appears to only happen when the file or symbol handler's `getResults` method (in the `this.openSymbolHandler.getResults(searchValue)` call) returns a synchronously resolved promise (which is not the case in VS Code). If the promise is asynchronously resolved (which is the case for VS Code's file and symbol handlers), then it works as expected.

Our ISearchService at Sourcegraph (which runs modiifed VS Code) returns file matches synchronously, which is where we saw the issue. Wrapping its call of the promise completion handler in `setTimeout` to make it trivially async makes it work for us.

We'll submit a patch for this early next week.

**Comment by bpasero** [comment]
@nicksnyder @chrmarti while this seems to work, I am not sure the behaviour is now for the better compared to stable. 

In stable we had a unified sorting across file and symbol results so that the most relevant entry was always to the top:

![image](https://cloud.githubusercontent.com/assets/900690/26666133/a0625ef8-469e-11e7-8c8a-c48f1b7f53bc.png)

This is no longer the case in insiders where symbol results seem to be just appended at the end after all file results:

![image](https://cloud.githubusercontent.com/assets/900690/26666142/b34b029a-469e-11e7-9c30-55cb8e3e4b01.png)

The tricky part about getting this right is actually not so much showing results one after the other but to resort the items so that relevant entries are still to the top. And then on top of that, a user might have already navigated to a result in the file list and would not want to loose that selection when the symbol results kick in.

**Comment by chrmarti** [comment]
Ack, reverted for the approaching release. (Including the fix for #27638 which was introduced with these changes.)

@nicksnyder If you get a chance to look into the sorting problem, we would be happy to give this another try. Thanks!

**Comment by nicksnyder** [comment]
No problem. Will take that into consideration.

**Comment by nicksnyder** [comment]
I just put up a new PR for this issue that should address that problems you saw with the last PR.
https://github.com/Microsoft/vscode/pull/28751

**Comment by chrmarti** [comment]
I'll wait for @bpasero 's verdict.
